src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -1154,3 +1154,71 @@ def restore_storage_account(client, vault_base_url, file_path):
         data = file_in.read()
         return client.restore_storage_account(vault_base_url, data)
 # endregion
+
+
+# region private_endpoint
+def show_private_endpoint_connection(client, resource_group_name, vault_name, connection_name,
+                                     connection_id=None):  # pylint: disable=unused-argument
+    """Show details of a private endpoint connection associated with a Key Vault."""
+    return client.get(resource_group_name=resource_group_name, vault_name=vault_name,
+                      private_endpoint_connection_name=connection_name)
+
+
+def delete_private_endpoint_connection(client, resource_group_name, vault_name, connection_name,
+                                       connection_id=None):  # pylint: disable=unused-argument
+    """ Delete the specified private endpoint connection associated with a Key Vault."""
+    return client.delete(resource_group_name=resource_group_name, vault_name=vault_name,
+                         private_endpoint_connection_name=connection_name)
+
+
+def approve_private_endpoint_connection(cmd, client, resource_group_name, vault_name, connection_name,
+                                        approval_description=None,
+                                        connection_id=None):  # pylint: disable=unused-argument
+    """Approve a private endpoint connection request for a Key Vault."""
+
+    PrivateEndpointConnection = cmd.get_models('PrivateEndpointConnection', resource_type=ResourceType.MGMT_KEYVAULT)
+    PrivateLinkServiceConnectionState = cmd.get_models('PrivateLinkServiceConnectionState',
+                                                       resource_type=ResourceType.MGMT_KEYVAULT)
+
+    private_endpoint_connection = client.get(resource_group_name=resource_group_name, vault_name=vault_name,
+                                             private_endpoint_connection_name=connection_name)
+
+    return client.put(resource_group_name=resource_group_name,
+                      vault_name=vault_name,
+                      private_endpoint_connection_name=connection_name,
+                      private_endpoint=private_endpoint_connection.private_endpoint,
```

Comment:
`private_endpoint` has been moved down to `properties` due to the parameter count change.  This is caused by autorest's `payload-flattening-threshold=2` config which will flatten the parameters when the parameter count <= 2. 

Anyway, let's remove it.